internal/agent/hooks/hook.go

```diff
@@ -36,10 +36,14 @@ var AfterUkiInstall = []Interface{
 }
 
 // AfterUkiReset sets which Hooks to run after uki runs the install action
-var AfterUkiReset = []Interface{}
+var AfterUkiReset = []Interface{
+	&Lifecycle{},
+}
 
 // AfterUkiUpgrade sets which Hooks to run after uki runs the install action
-var AfterUkiUpgrade = []Interface{}
+var AfterUkiUpgrade = []Interface{
+	&Lifecycle{},
```

Comment:
this was forgotten, so uki upgrade would not reset? have to check this further becuase I think we are doing this manually in the upgrade action which is bad

Reply:
yes, indeed, we were manually rebooting/poweroff-ing, probably becuase the lifecycle didnt hit

Fixed this plus got the hook together @mauromorales